### ISSUE TITLE
BelongsTo doesn't fetch and populate in the select dropdown 

### ISSUE THREAD

**Issue by AbanoubNassem** [laravel/nova-issues#2110]
- Laravel Version:  latest
- Nova Version: latest
- PHP Version: 7.3
- Database Driver and Version:  mysql
- Operating System and Version: mac
- Browser type and version: chrome

I have `BelongsToMany` relationship between `Restaurant` and `MenuItem`, it was working fine before upgrading to v2

`Restaurant Model:`

```
public function menuItems()
    {
        return $this->hasMany(MenuItem::class);
    }
```
`MenuItem Model`

 ```
   public function restaurant()
    {
        return $this->belongsTo(Restaurant::class);
    }
```

`Restaurant Resource:`

`HasMany::make('Menu Items', 'menuItems', 'App\Nova\MenuItem')->hideFromIndex(),`

`MenuItem Resource`

 `BelongsTo::make('Restaurant', 'restaurant', 'App\Nova\Restaurant'),`

now when I try to create a new menu item , the `Restaurant` dropdown is empty .

and sometimes I get this error in the console :

![Screen Shot 2019-11-05 at 4 05 57 PM](https://user-images.githubusercontent.com/13698722/68206559-727f0a00-ffe6-11e9-993c-dd3965e718de.png)

**Comment by AbanoubNassem** [comment]
alright , I know now , what is wrong, after hours of searching and digging , I found it errors out because of memory limit issue , since I have lots of restaurants in the database 

` local.ERROR: Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) `

now that , shows another issue!  and I was wondering about since I used `Nova` , which is imaging I have `100000` record of a model that I have a belongs to relationship with , now it doesn't make any sense to be a dropdown list , it should be a searchable select box :- benfest is it wont load the whole records only the one am searching for , hence no memory overload , and easier to use.

any suggestions how would I tackle this for now?
